src/silx/gui/plot/LegendSelector.py

```diff
@@ -148,6 +149,7 @@ class LegendModel(qt.QAbstractListModel):
     showLineRole = qt.Qt.UserRole + 3
     iconSymbolRole = qt.Qt.UserRole + 4
     showSymbolRole = qt.Qt.UserRole + 5
+    itemRole = qt.Qt.UserRole + 6
```

Comment:
Add a role to pass the curve item in the model and to operate directly on the item.
The goal is to make the legend widget works with curves that might have the same legend.

There is plenty of room to refactor the legend widget since the item provides all the other required information, but it would require to break the API and rewrite a lot of this file, so not for now.